asyncio_mqtt/client.py

```diff
@@ -302,67 +387,91 @@ def _on_disconnect(self, client, userdata, rc, properties=None):
+    def _on_unsubscribe(
+        self,
+        client: mqtt.Client,
+        userdata: Any,
+        mid: int,
+        properties: Optional[mqtt.Properties] = None,
+        reasonCodes: Optional[List[mqtt.ReasonCodes]] = None,
+    ) -> None:
         try:
-            self._pending_calls.pop(mid).set()
+            self._pending_unsubscribes.pop(mid).set()
         except KeyError:
             MQTT_LOGGER.error(
                 f'Unexpected message ID "{mid}" in on_unsubscribe callback'
             )
 
-    def _on_publish(self, client, userdata, mid):
+    def _on_publish(self, client: mqtt.Client, userdata: Any, mid: int) -> None:
         try:
-            self._pending_calls.pop(mid).set()
+            self._pending_publishes.pop(mid).set()
         except KeyError:
             # Do nothing since [2] may call on_publish before it even returns.
             # That is, the message may already be published before we even get a
             # chance to set up the 'pending_call' logic.
             pass
 
-    def _on_socket_open(self, client, userdata, sock):
-        def cb():
+    def _on_socket_open(
+        self, client: mqtt.Client, userdata: Any, sock: socket.socket
+    ) -> None:
+        def cb() -> None:
             client.loop_read()
 
         self._loop.add_reader(sock.fileno(), cb)
         # paho-mqtt calls this function from the executor thread on which we've called
         # `self._client.connect()` (see [3]), so we create a callback function to schedule
         # `_misc_loop()` and run it on the loop thread-safely.
-        def create_task_cb():
+        def create_task_cb() -> None:
             self._misc_task = self._loop.create_task(self._misc_loop())
 
         self._loop.call_soon_threadsafe(create_task_cb)
 
-    def _on_socket_close(self, client, userdata, sock):
+    def _on_socket_close(
+        self, client: mqtt.Client, userdata: Any, sock: socket.socket
+    ) -> None:
         self._loop.remove_reader(sock.fileno())
         if self._misc_task is not None:
             with suppress(asyncio.CancelledError):
                 self._misc_task.cancel()
 
-    def _on_socket_register_write(self, client, userdata, sock):
-        def cb():
+    def _on_socket_register_write(
+        self, client: mqtt.Client, userdata: Any, sock: socket.socket
+    ) -> None:
+        def cb() -> None:
             client.loop_write()
 
         self._loop.add_writer(sock, cb)
 
-    def _on_socket_unregister_write(self, client, userdata, sock):
+    def _on_socket_unregister_write(
+        self, client: mqtt.Client, userdata: Any, sock: socket.socket
+    ) -> None:
         self._loop.remove_writer(sock)
 
-    async def _misc_loop(self):
+    async def _misc_loop(self) -> None:
         while self._client.loop_misc() == mqtt.MQTT_ERR_SUCCESS:
             await asyncio.sleep(1)
 
-    async def __aenter__(self):
+    async def __aenter__(self) -> "Client":
```

Comment:
Alternatively, use `from __future__ import annotations`. Just check if that's Python 3.7 compatible beforehand. I can't remember. :)

Reply:
Yeah, I do tend to, but I've been writing 3.6+ code lately. What about this project is 3.7+ only? I ask because I've used it in my 3.6+ project :facepalm:

Reply:
I just checked to be sure: [`from __future__ import annotations` seems to be a Python 3.7 thing](https://docs.python.org/3/whatsnew/3.7.html#pep-563-postponed-evaluation-of-annotations). :)

> What about this project is 3.7+ only?

I think asyncio-mqtt (mostly) Python 3.6 compatible thanks to the efforts of pallas. I haven't kept track lately, though, so I might have broken 3.6 compatibility somewhere. 😅